File: CSharpFundamentals/CSharpOOPbasics/PolymorphismEx/WildFarm/Foods/Factory/FoodFactory.cs
﻿using System;

namespace WildFarm.Foods.Factory
{
    public class FoodFactory
    {
        public Food CreateFood(string type, int quantity)
        {
            type = type.ToLower();
            switch (type)
            {
                case "vegetable":
                    return new Vegetable(quantity);
                case "fruit":
                    return new Fruit(quantity);
                case "meat":
                    return new Meat(quantity);
                case "seeds":
                    return new Seeds(quantity);
                default:
                    return null;
            }
        }
    }
}

File: JavaFundamentals/JavaOOP/Polymorphism-Ex/src/wild_farm/FoodFactory.java
package wild_farm;

public class FoodFactory {
    public Food getFood(String[] tokens) {
        String type = tokens[0];
        Integer quantity = Integer.valueOf(tokens[1]);
        switch (type) {
            case "Vegetable":
                return new Vegetable(quantity);
            case "Meat":
                return new Meat(quantity);
        }
        return null;
    }
}
